packages/rabbitmq/src/amqp/connection.ts

```diff
@@ -343,17 +343,19 @@ export class AmqpConnection {
     originalHandlerName: string
   ): Promise<SubscriptionResult> {
     return new Promise((res) => {
-      this.selectManagedChannel(msgOptions?.queueOptions?.channel).addSetup(
-        async (channel) => {
+      this.selectManagedChannel(msgOptions?.queueOptions?.channel)
+        .addSetup(async (channel) => {
           const consumerTag = await this.setupSubscriberChannel<T>(
             handler,
             msgOptions,
             channel,
             originalHandlerName
           );
           res({ consumerTag });
```

Comment:
This worked when I applied it locally. Consider only resolving the `Promise` in one place. See one suggestion in the [comment](https://github.com/golevelup/nestjs/pull/495/files#r966322388) below.

Reply:
Hi @davidtemplin the addSetup functions receives a callback as a parameter and returns a promise. If the connection is successful it calls the callback function we passed. And if the connection fails it returns some value. So I think we would have to resolve or reject in both the callback function and then block.

Reply:
Thank you, @vijay94. Perhaps I have misunderstood. The `addSetup` function is called without specifying any `done` parameter. Then, `addSetup` returns the result of calling `pb.addCallback` without any `done` parameter. According to the documentation of `pb.addCallback`, this simply returns the supplied `Promise`, namely, this `Promise` (let's call it "Promise A"):

```typescript
(this._settingUp || Promise.resolve()).then(() => {
                this._setups.push(setup);
                if (this._channel) {
                    return pb.call(setup, this, this._channel);
                } else {
                    return undefined;
                }
            })
```

Promise A proceeds as follows: first, Promise A waits for the `Promise` `(this._settingUp || Promise.resolve())` to resolve, next the `then` handler is called, and finally Promise A resolves.

Now, when the `then` handler is called, the `setup` callback will be called if and only if `this._channel` is defined. However, _in all cases_, Promise A resolves to `undefined` (at least at the type level, i.e. assuming that the `setup` callback really is a `void` callback). If the `then` handler invokes `return pb.call`, it will wait for the `Promise` returned from `pb.call` to resolve to a value, and then resolve to the same value (which should be `undefined` since the `setup` callback is supposed to be a `void` function). If the `then` handler immediately returns `undefined`, then `Promise A` resolves immediately - but regardless, it resolves to `undefined`; the only difference is whether or not the `setup` callback is invoked. Either way, when it is done, it resolves to `undefined`.

Thus, the patch that I suggested should work. When Promise A is finished, the final `then` handler is called. If the `setup` callback was invoked, a value will be set in the `result` variable. Otherwise, `result` will be `undefined`.

Reply:
Yeah @davidtemplin, I went through the documentations and your solution will work. Please feel free to push the code directly to the master.

Reply:
Thanks, @vijay94 ! I'll do that soon.

Reply:
@vijay94 - branch protection is enabled for the `master` branch of the forked repo. Would you like me to create a PR with a new branch or would you prefer to disable the branch protection?

Reply:
Hi @davidtemplin, Can you raise a PR. I will approve and merge it right away. Thanks